nx/lib/nx.ex

```diff
@@ -9377,6 +9377,112 @@ defmodule Nx do
     if opts[:keep_axis], do: res, else: squeeze(res, axes: [axis])
   end
 
+  @doc """
+  Integrate `y` along the given axis using the composite trapezoidal rule.
+  The integration happens in sequence along elements of `x`.
+
+  ## Parameters
+
+      * `axis` - Axis along which to integrate. Defaults to `-1`.
+
+  ## Examples
+
+      iex> Nx.trapz(Nx.tensor([1, 2, 3]), Nx.tensor([4, 5, 6]))
+      #Nx.Tensor<
+        f32
+        4.0
+      >
+
+      iex> Nx.trapz(Nx.tensor([[0, 1, 2], [3, 4, 5]]), Nx.tensor([[1, 2, 3], [1, 2, 3]]))
+      #Nx.Tensor<
+        f32[2]
+        [2.0, 8.0]
+      >
+
+      iex> Nx.trapz(Nx.tensor([[0, 1, 2], [3, 4, 5]]), Nx.tensor([[1, 1, 1], [2, 2, 2]]), axis: 0)
+      #Nx.Tensor<
+        f32[3]
+        [1.5, 2.5, 3.5]
+      >
+  """
+  @doc type: :aggregation, from_backend: false
+  def trapz(y, x, opts \\ []) do
+    opts = keyword!(opts, axis: -1)
+    %T{shape: y_shape, names: y_names} = y = to_tensor(y)
+    %T{shape: x_shape} = x = to_tensor(x)
+
+    axis = Nx.Shape.normalize_axis(y_shape, opts[:axis], y_names)
+
+    d =
+      if rank(x) == 0 do
+        x_diff = diff(x)
```

Comment:
Isn't diff on a scalar 0?

Reply:
I don't think it makes sense to support rank 0 for x given that the trapezoidal rule needs 2 points for the calculation

Reply:
There should be some kind of compatibility check between x and y as well

Reply:
There should be == 1